cmake/modules/FindQAT.cmake

```diff
@@ -0,0 +1,40 @@
+find_package(PkgConfig)
+pkg_search_module(PC_QAT libqat qatlib QUIET)
+
+find_path(QAT_INCLUDE_DIR
+  NAMES qat/cpa.h
+  HINTS ${PC_QAT_INCLUDE_DIRS})
+
+find_library(QAT_LIBRARY
+  NAMES qat qat_s
```

Comment:
Here is the thing, if users' system isn't rhel>9 and their QAT devices are earlier than gen 4 devices, they need to install [out-of-tree QAT driver)](https://www.intel.com/content/www/us/en/download/19734/intel-quickassist-technology-driver-for-linux-hw-version-1-x.html). Becasue QAT gen 4 devices support qatlib, and earlier generations of QAT devices (e.g. c62x, dh895xxcc, etc.) are not supported by in-tree qat driver and [qatlib](https://github.com/intel/qatlib). By installing  out-of-tree QAT driver, there is no `qat/cpa.h` in system.

Reply:
i see, thanks. so the `FindQatDrv.cmake` file i replaced was specific to the out-of-tree build

the qatlib version is probably(?) the only one that's reasonable to package/ship to users by default, but we can preserve the ability to build with `FindQatDrv.cmake`. for example, `WITH_QAT` would build against qatlib by default, but you could select the other with `WITH_QATDRV` and crypto/qat/qcccrypto.h would decide which headers to include based on `#ifdef HAVE_QATDRV`

does that sound ok?

Reply:
great. Coincidentally, I have also been adding the function to build qatlib and QATzip automatically when compile Ceph in the past week and I'm almost done with the test. The main differences are: 1.  `WITH_QAT=ON` would find out-of-tree qat driver in system, which is basically unchanged ; `WITH_QATLIB` would build qatlib by Ceph compilation, which is basically the same as your `BuildQAT.cmake`. But I also **added a scene to install qatlib in [standalone mode](https://github.com/intel/qatlib/blob/abe15d7bfc083117bfbb4baee0b49ffcd1c03c5c/INSTALL#L496C5-L496C5) which is essential for Ceph to run on containers** 2. `WITH_QATZIP=ON` would first find out whether there is an installed QATzip library in the system, and if not, it will pull [QATzip code](https://github.com/intel/QATzip) and install it, which  is similar with your `BuildQATzip.cmake`, but distinguished between _out-of-tree qat driver_ and qatlib.
Your basic framework is more complete than mine. But IMHO, functionally, support for out-of-tree qat drver(i.e. support for earlier generations of QAT devices, e.g. c62x, dh895xxcc, etc.) cannot be deleted.

Reply:
that is a happy coincidence :)

how would you like to collaborate on this? you're welcome to take any of these changes into a branch/pr of your own. or i can adapt what i have here to use WITH_QAT/WITH_QATLIB/QAT_QATZIP as you've described, and we build on that

Reply:
Thanks so much. Your basic framework is very complete. I would like to make some changes based on your branch, mainly including installation compatibility of out-of-tree QAT dirver and the enablement of containerized ceph scene.